### PR TITLE
Add an initial kind-based end-to-end test suite

### PR DESCRIPTION
**Pull request by erwinvaneyk**
This PR adds an initial test suite that builds, deploys, and tests Decco on a Kubernetes cluster using KinD.

It is not as complete as the existing smoke tests: it does not deploy a Space or an App yet, nor does it integrate with external systems such as DNS management. However, it checks for the basics (building the executable, building the image, deploying Decco, verifying that it is Running and, checking if it creates the Space CRD). Also, on our CI systems, it runs in about 3-4 minutes.

### CODE REVIEW

**Comment by erwinvaneyk** [comment]
@pacharya-pf9 could you have a look too? Since you have been working with go testing/ginkgo a lot too if I remember correctly 😄

File: test/e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+func TestDecco(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Decco E2E Suite")
+}
+
+const (
+	testStagingDir            = "./test_staging"
+	testStagingManifestsDir   = testStagingDir + "/manifests"
+	testDataDir               = "./test_data"
```
Comment (pacharya-pf9) [inline test/e2e_test.go:26]: You can precreate a `./testdata` dir and it won't be included in the builds. 
https://medium.com/rungo/unit-testing-made-easy-in-go-25077669318 (look for Test Data section). Used it heavily in nodelet to hold files that I will use during unit tests.
You can then avoid all makedirs...

File: test/e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+func ensureCommandExists(name string) {
+	By("Checking if prerequisite command exists: " + name)
+	_, err := execCommand("which", name)
+	Expect(err).To(BeNil(), "Could not find prerequisite command '%s'", name)
+}
+
+func stageManifest(manifestPath string) {
+	By("Staging manifest: " + manifestPath)
+	out, err := execCommand("cp", manifestPath, testStagingManifestsDir)
```
Comment (pacharya-pf9) [inline test/e2e_test.go:209]: same as above. You can keep these files pre-created in testdata as well :). That trick is very handy

File: test/e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+			"--image", fmt.Sprintf("kindest/node:%s", testClusterVersion),
+			"--wait", "5m")
+		Expect(err).To(BeNil(), "Failed to create test KinD cluster: %s", string(out))
+
+		By("Setting KUBECONFIG to the kubeconfig of the test cluster")
+		err = os.Setenv("KUBECONFIG", testClusterKubeconfigPath)
+		Expect(err).To(BeNil(), "Failed to set KUBECONFIG")
+
+		By("Ensuring that the test cluster is accessible")
+		out, err = execCommand("kubectl", "version")
```
[the remark this answers is not in the document]
Comment (erwinvaneyk) [reply test/e2e_test.go:82]: From my personal experience, `kind create cluster` is itself a synchronous cluster creation command (likely with some checks of its own). Once it completes the cluster is fully functional. I will add some retrying logic just to be sure 👍

File: test/e2e_test.go (hunk 1 above)
Comment (erwinvaneyk) [reply test/e2e_test.go:26]: Thanks @pacharya-pf9 - I will take a look 🙂

File: test/e2e_test.go (hunk 2 above)
Comment (erwinvaneyk) [reply test/e2e_test.go]: There are some resources that I pull directly from `../manifests`. I copy them over to a temporary "test staging" directory to get an intermediate debuggable view of which exact resources will get installed in the test cluster (and to do so with a single `kubectl apply -f ${testManifestsDir}`. So it is a bit verbose, but that is intentional. ;)

File: test/e2e_test.go (hunk 1 above)
Comment (erwinvaneyk) [reply test/e2e_test.go]: I changed test_data to testdata and moved all permanent and temporary test data into that directory

**Pull request merged**